app/controllers/api/v1/sessions_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Api
+  module V1
+    class SessionsController < ApplicationController
+      skip_before_action :verify_authenticity_token
+      def create
+        auth = request.env["omniauth.auth"]
+        user = authenticate_with_omniauth(auth) if auth
+        
+        if user
+          if session[:user_id] == user.id
+            render json: { error: 'Already logged in' }, status: 422
```

Comment:
Is this desired?  422 falls under a failure range, and "already logged in" doesn't seem like a failure to me, particularly since they'll have to do the OAuth flow to get to this point.